### DISCUSSION TITLE
AnimationPlayer is lacking a method to play from frame x to frame y

### DISCUSSION THREAD

**Discussion by AndHalfAFish** [godotengine/godot-proposals#4294] [Engine Core]
The animationplayer has many methods, but the one I really miss is the one that lets me play part of an animation. 
I know there are ways around, but I imagine a method like this could be convenient for many Godot users : 

void play(name: String, from_frame: int, to_frame: int)

void play(name: String, from_frame: int, number_of_frames: int)

**Comment by Calinou** [comment]
AnimationPlayer does not think in terms of "frames" – it sees time as a continuous value in seconds.

Therefore, this can already be achieved with a Timer node or with a custom counter variable in `_process()`/`_physics_process()`.

**Comment by AndHalfAFish** [comment]
Yes, I've noticed that the animationplayer doesn't behave in the way I expect it to behave (from experience with other players). I know there are workarouonds, but it isn't convenient if you have to do these things many times in a project.
Anyway, one can only ask ... =)

**Reply by AndHalfAFish** [reply]
I just found this in the documentation, and I wondered why Godot would offer this functionality for imported animations, but not for native animations when game developers obviously feel a need for it ? 

https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_scenes.html#clip
![animationplayer_clips](https://user-images.githubusercontent.com/59080934/161911958-f9565fef-d5fd-4536-a0bf-ce386f10fac1.png)
